data/data/agent/files/usr/local/bin/load-config-iso.sh

```diff
@@ -0,0 +1,67 @@
+#!/bin/bash
+
+set -e
+
+AGENT_CONFIG_ARCHIVE_FILE="config.gz"
+AGENT_CONFIG_MOUNT="/mnt"
+RENDEZVOUS_HOST_ENV="/etc/assisted/rendezvous-host.env"
+CLUSTER_IMAGE_SET="/etc/assisted/manifests/cluster-image-set.yaml"
+
+copy_archive_contents() {
+ 
+    tmpdir=$(mktemp --tmpdir -d "config-image--XXXXXXXXXX")
+    cp ${AGENT_CONFIG_MOUNT}/${AGENT_CONFIG_ARCHIVE_FILE} "${tmpdir}"
+    gunzip -f ${tmpdir}/${AGENT_CONFIG_ARCHIVE_FILE}
+    unzipped_file=$(echo "${tmpdir}/${AGENT_CONFIG_ARCHIVE_FILE}" | cut -d'.' -f1)
+
+    # Get the releaseImage in the archive and verify it matches the current cluster-image-set
+    release_image=$(grep releaseImage ${CLUSTER_IMAGE_SET} | sed -n -e 's/^.*releaseImage: //p')
+    arch_image_set=$(cpio -icv --to-stdout ${CLUSTER_IMAGE_SET} < "${unzipped_file}")
+    if [[ $(< "${CLUSTER_IMAGE_SET}") != "${arch_image_set}" ]]; then
+       echo "The cluster-image-set in archive does not match ${CLUSTER_IMAGE_SET}"
+       cleanup_files
+       return 1
+    fi
+    echo "Archive on ${device} contains release ${release_image}"
+
+    # Copy all files from archive
+    # The rendezvousIP file must be copied last as it triggers set-node-zero.sh to continue configuration
+    cpio -icvd --nonmatching "${AGENT_RENDEZVOUS_IP_FILE}" < "${unzipped_file}"
+    cpio -icvd "${RENDEZVOUS_HOST_ENV}" < "${unzipped_file}"
+
+    echo "Successfully copied contents of ${AGENT_CONFIG_ARCHIVE_FILE} on ${device}"
+
+    cleanup_files
+    return 0
+}
+
+cleanup_files() {
+
+    if [[ -f "${unzipped_file}" ]]; then
+       rm "${unzipped_file}"
+    fi
+    if [[ -d "${tmpdir}" ]]; then
+       rmdir "${tmpdir}"
+    fi
+}
+
+# This script will be invoked by a udev rule when it detects a device with the correct label
+devname="$1"
+systemd-mount "${devname}" "${AGENT_CONFIG_MOUNT}"
+echo "Mounted ${devname} on ${AGENT_CONFIG_MOUNT}"
+
+while true
+do
+    if [[ -f ${AGENT_CONFIG_MOUNT}/${AGENT_CONFIG_ARCHIVE_FILE} ]]; then
+       # Copy contents from archive
+       if copy_archive_contents; then
+          break
+       fi
+    else
+       echo "Could not find ${AGENT_CONFIG_ARCHIVE_FILE} in ${AGENT_CONFIG_MOUNT}"
+    fi
+
+    echo "Retrying to copy contents from ${AGENT_CONFIG_MOUNT}/${AGENT_CONFIG_ARCHIVE_FILE}"
+    sleep 5
+done
+
```

Comment:
Because the unconfigured ignition disables these services:
* apply-host-config.service
* start-cluster-installation.service

They should be enabled here.

Reply:
OK I will add these

Reply:
@rwsu Actually could we add those services in the unconfigured ignition with a dependency on the load-config-iso.service

Reply:
I like that idea, assuming it can be templated in to the unit easily enough.
I too was expecting that we would enable the services here, but calling systemctl from inside a systemd service is definitely a bit icky.

Reply:
Actually, does this work when we can't predict the service name (due to it being templated with the device path, which we can't predict)?
And did this change actually happen somewhere?

Reply:
I actually don't think that any additional service dependencies are needed. Since `set-node-zero.sh` is looping waiting for the  rendezvous host file to appear, when it does the `node-zero.service` completes, which will allow the `assisted-service-pod.service` (https://github.com/openshift/installer/blob/master/data/data/agent/systemd/units/assisted-service-pod.service#L4) to run, followed by create-cluster-and-infraenv.service

Reply:
That's true for the appliance if `apply-host-config.service` and `start-cluster-installation.service` are enabled in the unconfigured ignition.
But if we want the unconfigured ignition to work for an interactive workflow, we need to _disable_ those services by default and only enable them if the config drive is attached.
I think the only way to do this is to call systemctl, and we might as well implement it now.

Reply:
Added ability to enable the disabled services in `load-config-iso.sh`